pkg/netpol/eval/internal/k8s/netpol.go

```diff
@@ -154,20 +165,25 @@ func (np *NetworkPolicy) ruleConnections(rulePorts []netv1.NetworkPolicyPort, ds
 			// 1. empty-numbered-range with the string namedPort, if the rule has a named-port which is not (or cannot be) converted to a
 			// numbered-range by the dst's ports.
 			//  2. non empty-range when the rule ports are numbered or the named-port was converted
-			if (dst == nil || isPeerRepresentative(dst)) && isEmptyPortRange(startPort, endPort) && portName != "" {
+			if isEmptyPortRange(startPort, endPort) && portName != "" {
 				// this func may be called:
 				// 1- for computing cluster-wide exposure of the policy (dst is nil);
 				// 2- in-order to get a connection from a real workload to a representative dst.
 				// in both first cases, we can't convert the named port to its number, like when dst peer is a real
 				// pod from manifests, so we use the named-port as is.
 				// 3- in order to get a connection from any pod to a real dst.
 				// in the third case the namedPort of the policy rule may not have a match in the Pod's configuration,
-				// so it will be ignored (the pod has no matching named-port in its configuration - unknown connection is not allowed)
+				// so it will be ignored and a warning is raised
+				// (the pod has no matching named-port in its configuration - unknown connection is not allowed)
 				// 4- in order to get a connection from any pod to an ip dst (will not get here, as named ports are not defined for ip-blocks)
-
-				// adding portName string to the portSet
-				ports.AddPort(intstr.FromString(portName))
-			}
+				if dst == nil || isPeerRepresentative(dst) { // (1 & 2)
+					// adding portName string to the portSet
+					ports.AddPort(intstr.FromString(portName))
+				} else { // dst is a real pod (3)
+					// raising a warning that the "named port" of the rule is ignored, since it has no match in the pod config.
+					np.Logger.Warnf(np.netpolWarning(alerts.WarnUnmatchedNamedPort(portName, dst.String())))
+				}
+			} // @tbd should raise a warning/error if the portRange is empty (when also namedPort is empty)
```

Comment:
as for "@tbd should raise a warning/error if the portRange is empty" :
please add an example test covering this case.. what is the output currently? can a warning by sufficient? (rule becomes ineffective?)

Reply:
added a test.
currently an empty port range is ignored (skipped); 
so on one hand I can add a warning (since no problems are caused); but on the other hand according to [this](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.31/#networkpolicyport-v1-networking-k8s-io)
end port: 

> This field cannot be defined if the port field is not defined or if the port field is defined as a named (string) port. The endPort must be equal or greater than port

.
so it is illegal; does this mean we should raise an error?

Reply:
ok, so let's change to an error

Reply:
done - for anp too

Reply:
done